### PR TITLE
Fix issues from review for the Kptncook scraper

### PR DESCRIPTION
**Pull request by gloriousDan**
This PR aims to resolve the comments raised by @jayaddison in PR #503  which introduced the Kptncook scraper

### CODE REVIEW

**Comment by gloriousDan** [comment]
I'm not sure why the coveralls job fails, is that caused by the changes in my PR? @jayaddison

File: recipe_scrapers/kptncook.py [hunk 1]
```diff
@@ -16,12 +15,7 @@
 class KptnCook(AbstractScraper):
     def __init__(self, url, *args, **kwargs):
         super().__init__(url, *args, **kwargs)
-        if urlparse(url).hostname == "mobile.kptncook.com":
-            parsed_url = urlparse(url)
-        else:
-            # If it's a sharing.kptncook.com/* link we first need to follow the http forwards to get the recipe ID
-            recipe_request = requests.get(url, headers=HEADERS)
-            parsed_url = urlparse(recipe_request.url)
+        parsed_url = urlparse(self.url)
```
Comment (jayaddison) [inline recipe_scrapers/kptncook.py:18]: Neat - this seems a lot simpler logically, and solves the duplicate-request problem.  Thank you!

Resolves https://github.com/hhursev/recipe-scrapers/pull/503#discussion_r901527643

**Comment by jayaddison** [comment]
> I'm not sure why the coveralls job fails, is that caused by the changes in my PR? @jayaddison

Me neither - [HTTP 405](https://github.com/hhursev/recipe-scrapers/runs/7129454047?check_suite_focus=true#step:7:53) is an unexpected response and kinda suggests to me that the coveralls API might have changed, or something like that.

A few moments ago I [re-ran the unit tests](https://github.com/hhursev/recipe-scrapers/actions/runs/2570842557) on our main branch, and although they passed, there is some [coveralls API-related error output](https://github.com/hhursev/recipe-scrapers/runs/7130921642?check_suite_focus=true#step:7:32) that appears.  I'll try re-running them for this pull request too.

**Comment by jayaddison** [comment]
This looks good to me, thanks @gloriousDan - and my apologies again for not catching more of these during the first review pass.  The one remaining item that would be nice to resolve, I think, would be to update the test data for the unit test `GET` fixture so that it contains the HTML content served on first-request.

**Comment by jayaddison** [comment]
> A few moments ago I [re-ran the unit tests](https://github.com/hhursev/recipe-scrapers/actions/runs/2570842557) on our main branch, and although they passed, there is some [coveralls API-related error output](https://github.com/hhursev/recipe-scrapers/runs/7130921642?check_suite_focus=true#step:7:32) that appears. I'll try re-running them for this pull request too.

Seems like they [passed](https://github.com/hhursev/recipe-scrapers/actions/runs/2589239654) this time around, so perhaps the previous failure in this pull request was due to a temporary issue with the `coveralls` service (they have [reported one or two service status issues](https://status.coveralls.io/#past-incidents) in the past 24h, although mostly performance-related).

Not-directly-related: the [HTTP 422 / unprocessable entity messages continue to appear](https://github.com/hhursev/recipe-scrapers/runs/7131037415?check_suite_focus=true#step:7:21), so I'll take a brief look into those.

**Comment by gloriousDan** [comment]
@jayaddison I really like the way requests can be mocked now, this makes testing complicated scrapers like this one much easier and the code can also be cleaner. And also thanks for your attention to detail and the thorough review :)

**Comment by jayaddison** [comment]
You're welcome!  I added one more comment/question about measurements in #503, but I don't know where we'll get to with that conversation, and I don't think it's a blocker for the improvements here, so I'll approve & merge soon - thank you.

**Comment by jayaddison** [comment]
PS: I didn't manage to figure out the HTTP 422 / unprocessable entity issue yesterday; I'll file an issue for that soon.

**Review by jayaddison — APPROVED** [review]

**Pull request merged**